lib/hammer_cli/command_extensions.rb

```diff
@@ -0,0 +1,222 @@
+      @help_extension_block = block
+    end
+
+    def self.request_headers(&block)
+      @request_headers_block = block
+    end
+
+    def self.request_options(&block)
+      @request_options_block = block
+    end
+
+    def self.request_params(&block)
+      @request_params_block = block
+    end
+
+    def self.option_sources(&block)
+      @option_sources_block = block
+    end
+
+    # Object
+
+    def extend_options(command_class)
+      allowed = @only & %i[command_options option]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_options(command_class)
+    end
+
+    def extend_before_print(data)
+      allowed = @only & %i[before_print data]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_before_print(data)
+    end
+
+    def extend_output(command_class)
+      allowed = @only & %i[output]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_output(command_class)
+    end
+
+    def extend_help(command_class)
+      allowed = @only & %i[help]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_help(command_class)
+    end
+
+    def extend_request_headers(headers)
+      allowed = @only & %i[request_headers headers request]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_request_headers(headers)
+    end
+
+    def extend_request_options(options)
+      allowed = @only & %i[request_options options request]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_request_options(options)
+    end
+
+    def extend_request_params(params)
+      allowed = @only & %i[request_params params request]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_request_params(params)
+    end
+
+    def extend_option_sources(sources, command = nil)
+      allowed = @only & %i[option_sources]
+      return if allowed.empty? || (allowed & @except).any?
+
+      self.class.extend_option_sources(sources, command)
+    end
+
+    def delegatee(command_class)
+      self.class.delegatee = command_class
+    end
+
+    def details
+      except = @except.empty? ? '*nothing*' : @except
+      details = if @only == ALLOWED_EXTENSIONS
+                  "*all* except #{except}"
+                else
+                  "#{@only} only"
+                end
+      "#{self.class} for #{details}"
+    end
+
+    # Class
+
+    def self.extend_options(command_class)
+      return if @options.nil?
+
+      @options.each do |option|
+        command_class.send(:option,
+                           option[:switches],
+                           option[:type],
+                           option[:description],
+                           option[:opts],
+                           &option[:block])
+        logger.debug("Added option for #{command_class}: #{ApipieBindings::Utils.inspect_data(option)}")
```

Comment:
@ofedoren, ` #{ApipieBindings::Utils.inspect_data(option)}` feels quite 'aggressive' on space. With the environments PR I got most of the debug out filled with the extensions only in random command. I tend to prefer the less readable  `#{option}` here. What do you think?

Reply:
I totally agree on that one :)